pkg/client/edv/client.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package edv
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/trustbloc/edv/pkg/restapi/edv/operation"
+)
+
+// Client is used to interact with an EDV server.
+type Client struct {
+	edvServerURL string
+}
+
+// New returns a new instance of an EDV client.
+func New(edvServerURL string) *Client {
+	return &Client{edvServerURL: edvServerURL}
+}
+
+// CreateDataVault sends the EDV server a request to create a new data vault.
+// The location of the newly created data vault is returned.
+func (c *Client) CreateDataVault(config *operation.DataVaultConfiguration) (string, error) {
+	jsonToSend, err := json.Marshal(config)
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := http.Post(c.edvServerURL+"/data-vaults", "application/json", bytes.NewBuffer(jsonToSend))
```

Comment:
duplicate code

Reply:
Done .(I had to combine `jsonToSend, err := json.Marshal(config)` and `resp, err := http.Post(c.edvServerURL+"/data-vaults", "application/json", bytes.NewBuffer(jsonToSend))` along with their error checks in order to bury the uncovered line into a common area, though).

Also keep in mind that I still have to check for the error where I use that function, so I'm probably making the PR bigger or around the same size